pkg/aws/services/vpclattice.go

```diff
@@ -169,3 +177,65 @@ func (d *defaultLattice) ListServiceNetworkServiceAssociationsAsList(ctx context
 
 	return result, nil
 }
+
+func (d *defaultLattice) FindServiceNetwork(ctx context.Context, name string, optionalAccountId string) (*ServiceNetworkInfo, error) {
+	input := vpclattice.ListServiceNetworksInput{}
+
+	for {
+		resp, err := d.ListServiceNetworksWithContext(ctx, &input)
```

Comment:
there is implementation for paginated view in sdk: ListListenersPagesWithContext. That will replace sharp-edged infinite for loop with closure function.

https://github.com/aws/aws-sdk-go/blob/v1.45.6/service/vpclattice/vpclatticeiface/interface.go#L195

Reply:
I'll sync up with you offline on this one. I adapted existing implementations in this file, but if they are flawed then that's no good. Basically every other method in this file has this pattern.

Reply:
Using the paginated methods will really simplify the methods in this file. We agreed I could postpone that to the next PR.